src/crypto/crypto.ts

```diff
@@ -52,12 +53,26 @@ const keychainPromises = {
    * @param account The keychain account name.
    */
   getPassword(_keychain: KeyChain, service: string, account: string): Promise<CredType> {
-    return new Promise((resolve, reject): {} =>
-      _keychain.getPassword({ service, account }, (err: Nullable<Error>, password?: string) => {
-        if (err) return reject(err);
-        return resolve({ username: account, password: ensure(password) });
-      })
-    );
+    const sb = Cache.get<SecureBuffer<string>>(`${service}:${account}`);
+    if (!sb) {
+      return new Promise((resolve, reject): {} => {
+        return _keychain.getPassword({ service, account }, (err: Nullable<Error>, password?: string) => {
+          if (err) return reject(err);
+          const newSb = new SecureBuffer<string>();
+          newSb.consume(Buffer.from(ensure(password), 'utf8'));
+          Cache.set(`${service}:${account}`, newSb);
+          return resolve({ username: account, password: ensure(password) });
+        });
+      });
+    } else {
+      const pw = sb.value((buffer) => buffer.toString('utf8'));
```

Comment:
Extract password from SecureBuffer to pass back to caller
Store the password in cache in a new SecureBuffer